### PR TITLE
MAPI-1340 - lenient topic validation

### PR DESCRIPTION
**Pull request by agulowaty**
cc @pvighi @alexduf @DiegoVazquezNanini

### CODE REVIEW

File: registration/app/registration/services/topic/TopicValidator.scala [hunk 1]
```diff
@@ -17,11 +17,12 @@ import scalaz.syntax.either._
...
   def removeInvalid(topics: Set[Topic]): Future[TopicValidatorError \/ Set[Topic]]
 
-  trait TopicValidatorError {
-    def reason: String
-  
-    def topicsQueried: Set[Topic]
-  }
+}
+
+trait TopicValidatorError {
```
Comment (agulowaty) [inline registration/app/registration/services/topic/TopicValidator.scala]: made public to mock in Spec

File: registration/app/registration/controllers/Main.scala [hunk 2]
```diff
@@ -30,17 +30,27 @@ final class Main @Inject()(notificationRegistrarSupport: RegistrarSupport, topic
...
+    }
+
+    def registerWith(registrar: NotificationRegistrar) =
+      registrar
+        .register(lastKnownDeviceId, registration)
+        .map {processResponse}
 
     registrarFor(registration) match {
+      case \/-(registrar) =>
+        validate(registration.topics)
```
Comment (davidfurey) [inline registration/app/registration/controllers/Main.scala]: Should the return value of this be ignored?

Comment (agulowaty) [inline registration/app/registration/controllers/Main.scala]: yes, problems are logged inside method and we don't want to chain it with registration flow

Comment (pvighi) [inline registration/app/registration/controllers/Main.scala]: So we are not removing the invalid topics even if we get a correct response from auditors ? or am i missing something?

Comment (agulowaty) [inline registration/app/registration/controllers/Main.scala]: good point, fixed

**Comment by davidfurey** [comment]
:+1:

**Pull request merged**